src/main/java/io/vertx/core/impl/ContextImpl.java

```diff
@@ -181,9 +175,26 @@ public VertxInternal owner() {
         if (metrics != null) {
           metrics.end(execMetric, fut.succeeded());
         }
-        if (resultHandler != null) {
-          fut.setHandler(ar -> context.runOnContext(v -> resultHandler.handle(ar)));
-        }
+        fut.setHandler(ar -> {
```

Comment:
when the `resultHandler` is null, instead we should provide a default task that will log the error on the context, so something like:

```java
if (resultHandler == null) {
  resultHandler = ar -> {
    if (ar.failed) {
      context.reportException(ar.cause());
    }
  }
}
```

Reply:
Done